spec/std/enumerable_spec.cr

```diff
@@ -416,6 +416,11 @@ describe "Enumerable" do
     end
   end
 
+  describe "empty?" do
+    it { %w[foo bar].empty?.should be_false }
+    it { %w[].empty?.should be_true }
```

Comment:
This is testing `Indexable#empty?` since `Array` includes that.

Could just check `.each.empty?` for now since `Iterator` is `Enumerable` but not `Indexable`, but I think I would prefer a custom type including `Enumerable` for the test to future proof it from testing the wrong thing in case we add  `Iterator#empty?` for some reason.

Reply:
> This is testing `Indexable#empty?` since `Array` includes that.

That's correct, and it's a current situation for many other methods tested in the spec btw...

Reply:
You should use `SpecEnumerable` defined in that spec file.

Reply:
And possibly introduce another one to prove that `empty?` returns `true`. Maybe change `SpecEnumerable` to wrap an array and call `each` on it. That way testing is easier. And we could eventually use that throughout the whole spec file.

Reply:
That's what I ended up doing, see the force-pushed code.